### ISSUE TITLE
Weird side effect from `ctas_approach=True`

### ISSUE THREAD

**Issue by JPFrancoia** [awslabs/aws-data-wrangler#335]
Hi,

I just would like to mention a weird side effect that I encountered today.

I'm building a dataframe from an Athena table, like this:

```python
df: pd.DataFrame = wr.athena.read_sql_query(sql=query, database=clean_database)
```

`ctas_approach` is true by default.

Later on, I want to mutate one of the columns (which contains a datetime):

```python
df["date"] = df["date"] + pd.DateOffset(months=-2)
```

This should normally be totally valid pandas. But I get:

```
File "/Users/jpfrancoia/Library/Caches/pypoetry/virtualenvs/glue-jobs-KVdNkci1-py3.6/lib/python3.6/site-packages/pandas/core/ops/common.py", line 64, in new_method
    return method(self, other)
  File "/Users/jpfrancoia/Library/Caches/pypoetry/virtualenvs/glue-jobs-KVdNkci1-py3.6/lib/python3.6/site-packages/pandas/core/ops/__init__.py", line 503, in wrapper
    result = arithmetic_op(lvalues, rvalues, op, str_rep)
  File "/Users/jpfrancoia/Library/Caches/pypoetry/virtualenvs/glue-jobs-KVdNkci1-py3.6/lib/python3.6/site-packages/pandas/core/ops/array_ops.py", line 193, in arithmetic_op
    res_values = dispatch_to_extension_op(op, lvalues, rvalues)
  File "/Users/jpfrancoia/Library/Caches/pypoetry/virtualenvs/glue-jobs-KVdNkci1-py3.6/lib/python3.6/site-packages/pandas/core/ops/dispatch.py", line 125, in dispatch_to_extension_op
    res_values = op(left, right)
  File "/Users/jpfrancoia/Library/Caches/pypoetry/virtualenvs/glue-jobs-KVdNkci1-py3.6/lib/python3.6/site-packages/pandas/core/ops/common.py", line 64, in new_method
    return method(self, other)
  File "/Users/jpfrancoia/Library/Caches/pypoetry/virtualenvs/glue-jobs-KVdNkci1-py3.6/lib/python3.6/site-packages/pandas/core/arrays/datetimelike.py", line 1332, in __add__
    result = self._add_offset(other)
  File "/Users/jpfrancoia/Library/Caches/pypoetry/virtualenvs/glue-jobs-KVdNkci1-py3.6/lib/python3.6/site-packages/pandas/core/arrays/datetimes.py", line 689, in _add_offset
    result = offset.apply_index(values).tz_localize(self.tz)
  File "pandas/_libs/tslibs/offsets.pyx", line 114, in pandas._libs.tslibs.offsets.apply_index_wraps.wrapper
  File "/Users/jpfrancoia/Library/Caches/pypoetry/virtualenvs/glue-jobs-KVdNkci1-py3.6/lib/python3.6/site-packages/pandas/tseries/offsets.py", line 333, in apply_index
    shifted = liboffsets.shift_months(i.asi8, months)
  File "pandas/_libs/tslibs/offsets.pyx", line 745, in pandas._libs.tslibs.offsets.shift_months
  File "stringsource", line 658, in View.MemoryView.memoryview_cwrapper
  File "stringsource", line 349, in View.MemoryView.memoryview.__cinit__
ValueError: buffer source array is read-only
```

From what I could understand, it's somehow related to pyarrow and its immutability (I think, but I could be wrong): https://github.com/apache/arrow/issues/4128.

However, simply querying my dataframe like this completely solves the error:

```python
df: pd.DataFrame = wr.athena.read_sql_query(sql=query, database=clean_database, ctas_approach=False)
```

Is it expected ? (tested with awswrangler 1.6.3).

**Comment by igorborgest** [comment]
Hi @JPFrancoia! Good catch as usual!

The fix already was applied to our development branch, could you validate it please?

```sh
pip install git+https://github.com/awslabs/aws-data-wrangler.git@dev
```

**Comment by JPFrancoia** [comment]
Sorry for the delay. I confirm that my code now works with ctas_approach=True :) Good job

**Comment by igorborgest** [comment]
Thank you one more time @JPFrancoia

**Comment by igorborgest** [comment]
Released in [1.7.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/1.7.0)!

**Comment by igorborgest** [comment]
Hi @JPFrancoia !

It is totally off-topic, but we are stating a [_"Who uses AWS Data Wrangler?"_](https://github.com/awslabs/aws-data-wrangler/tree/dev#who-uses-aws-data-wrangler) section. So feel free to add yourself if you want 😄 .
